src/server/routes/api/search.ts

```diff
@@ -0,0 +1,69 @@
+import { Router } from "express";
+import db from "../../db";
+import patterns from "../../db/queries/patterns";
+import { IPatternTable, Tag, Tags } from "../../types";
+
+const router = Router();
+
+//GET /api/search/title/:param
+router.get("/title/:queryString", async (req, res, next) => {
+	let title = req.params.queryString;
+	try {
+		const result = await db.search.findByTitle(title);
+		if (result.length === 0) {
+			return res.status(404).json({ message: "No patterns found" });
+		}
+		res.json({ result, message: "Patterns found" });
+	} catch (error) {
+		console.error("Error fetching patterns:", error);
+		next(error);
+	}
+});
+
+//POST /api/search/tag
+router.post("/tag", async (req, res, next) => {
+	try {
+		const finalPatterns: IPatternTable[] = [];
+		let tags: Tags;
+
+		try {
+			tags = JSON.parse(req.body.tagList);
+		} catch (err) {
+			return res.status(400).json({ message: "Invalid JSON format" });
+		}
+		const idArray: number[] = tags.map((tag: Tag) => tag.id);
+
+		const patternPromises = idArray.map((id: number) => {
+			return db.search.findByTags(id);
+		});
+
+		//Zach what type is this??
+		let result;
+		try {
+			result = await Promise.all(patternPromises);
+		} catch (err) {
+			return res.status(500).json({ message: "Database error" });
+		}
+
+		if (result.length === 0) {
+			return res.json(404).json({ message: "No patterns found" });
+		}
+
+		console.log(`RESULT`, result);
+		const uniquePatterns = new Set();
```

Comment:
I see what you are doing with the Set, but I believe this can just be an array and you can use .includes to see if it has pattern.id in it further down. From what I just looked up, arrays are more performant than sets at smaller list sizes: https://stackoverflow.com/questions/39007637/javascript-set-vs-array-performance

Reply:
Changed to an array, this was a suggestion from ChatGPT and I liked it cause I learned something new! I'll leave a comment in the code that it can be converted to a set if the data ends up being more performant that way, and to remind myself there's another way to do that 😄